Lean4Axiomatic/Rational/Reciprocation.lean

```diff
@@ -183,6 +182,58 @@ variable {ℚ : Type}
   [Core (ℤ := ℤ) ℚ] [Addition ℚ] [Multiplication ℚ]
   [Negation ℚ] [Reciprocation ℚ] [Division ℚ]
 
+/--
+Equivalent to `Division.Props.ind_fraction` but with a more convenient argument
+order when using the `apply` tactic.
+-/
+def ind_fraction_on
+  {motive : ℚ → Prop}
+  [AA.Substitutive₁ (α := ℚ) (motive) (· ≃ ·) (· → ·)] (p : ℚ) 
+    (on_int_frac : ({a b : ℤ} → [Integer.Nonzero b] → motive ( (a : ℚ) / b)))
+    : motive p :=
+    ind_fraction on_int_frac p
+
+/--
+The predicate AsRatio satisfies a substitivie property with respect to the
+equivalence relation ≃. 
+I.e. If two rationals a and b are equvalent and a can be expressed as an
+integer ratio, a ≃ a' / b', then b can be expressed in the same way.
+-/
+theorem AsRatio_prop_subst_lemma : ∀ {a b : ℚ}, a ≃ b → AsRatio a → AsRatio b
+    := by
+  intro a b aeb as_ratio_a 
+  exact match as_ratio_a with
+  | AsRatio.intro a' b' bnz eqfrac =>
+    have b_as_ratio : b ≃ a' / b' := calc
+      b      ≃ _ := eqv_symm aeb
+      a      ≃ _ := eqfrac
+      from_integer a' / from_integer b'  ≃ _ := eqv_refl
+    AsRatio.intro a' b' bnz b_as_ratio
+
+instance AsRatio_subst : AA.Substitutive₁ (α := ℚ) (AsRatio) (· ≃ ·) (· → ·) := 
+{
+  subst₁ := AsRatio_prop_subst_lemma
+}
+
+  /--
+  Every rational number can be expressed as a ratio of integers.
+
+  Given any two integers, we can easily make a rational number; convert both of
+  them to rationals using `from_integer`, then divide them. This theorem tells us
+  that we can also do the reverse: given any rational, there are two integers
+  that produce it when put into a ratio.
+
+  It's a useful property because it provides a way to "deconstruct" a rational
+  number into simpler pieces, which may be easier to work with. Although, it's
+  preferable to work with rational numbers directly, and use this only when
+  necessary.
+  -/
+theorem rational_as_ratio (p : ℚ) : AsRatio p := by
```

Comment:
Any reason for renaming this from `as_ratio` to `rational_as_ratio`? I think the `rational_` prefix is a bit redundant, because the theorem lives in the `Lean4Axiomatic.Rational` namespace. If there's a context where there's a risk of name conflict, or just ambiguity, then this theorem can be referred to as `Rational.as_ratio`, which provides the same level of clarity.